aya/src/programs/fentry.rs

```diff
@@ -0,0 +1,78 @@
+//! fentry programs.
+use thiserror::Error;
+
+use crate::{
+    generated::{bpf_attach_type::BPF_TRACE_FENTRY, bpf_prog_type::BPF_PROG_TYPE_TRACING},
+    obj::btf::{Btf, BtfError, BtfKind},
+    programs::{load_program, utils::attach_btf_id, LinkRef, ProgramData, ProgramError},
+};
+
+#[derive(Debug)]
+#[doc(alias = "BPF_TRACE_FENTRY")]
+#[doc(alias = "BPF_PROG_TYPE_TRACING")]
+pub struct FEntry {
+    pub(crate) data: ProgramData,
+}
+
+#[derive(Debug, Error)]
+pub enum FEntryLoadError {
+    #[error(transparent)]
+    Btf(#[from] BtfError),
+
+    #[error(transparent)]
+    Program(#[from] ProgramError),
+}
+
+/// Marks a function as a fentry eBPF program that can be attached to almost
```

Comment:
Move this up to the struct definition please.

Also the format must be:

```
Title

Longer description
```

For example:

```
A program that can be attached to the entry point of (almost) any kernel function.

Fentry programs are similar to [kprobes](...), but have less overhead. Additionally fentry programs can also be attached to other eBPF programs.
```

Reply:
Done